.github/workflows/tests.yml

```diff
@@ -0,0 +1,39 @@
+name: Run tests
+
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: Ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run linters
+        run: tox -e lint
+        
+  unit-test:
```

Comment:
Add a similar stanza to this `unit-test` for integration tests. The tox target should be `tox -e integration` and the timeout very likely needs to be increased. Something similar to (untested):

```yaml
integration-test:
  name: Integration tests
  runs-on: ubuntu-latest
  timeout-minutes: 30
  steps:
    - name: Checkout source code
      uses: actions/checkout@v4
    - name: Install dependencies
      run: python -m pip install tox --user tox
    - name: Clear tox cache
      run: export TOX_WORK_DIR=$HOME/.tox
    - name: Update package lists
      run: sudo apt-get update
    - name: Run integration tests
      run: tox -e integration
```